.circleci/config.yml

```diff
@@ -214,46 +150,80 @@ build_and_deploy: &build_and_deploy
               echo "Creating new multidev site cloned from $BASE. Please wait."
               terminus multidev:create $CIRCLE_PROJECT_REPONAME.$BASE $PANTHEON_ENV
             fi
-
             rm -rf ./env_list.txt
-            # Set the connection type to GIT
-            terminus connection:set $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV git
-            # Force push $CIRCLE_BRANCH to Pantheon as branch named $PANTHEON_ENV
-            git push -f pantheon $CIRCLE_BRANCH:$PANTHEON_ENV
+          fi
 
-            echo "Waiting for sync code workflow to finish..."
-            until [ $(terminus workflow:list --fields status,env,workflow --format string -- $CIRCLE_PROJECT_REPONAME | egrep -i "$PANTHEON_ENV\s+Sync code" -m 1 | egrep -i -o "^\w+") == succeeded ]; do
-              sleep 30
-            done
-            echo "Rebuilding cache..."
-            terminus drush $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV -- cr
-            echo "Running database updates..."
-            terminus drush $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV -- updb -y
-            echo "Importing config..."
-            terminus drush $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV -- cim -y -vvv
-            echo "Rebuilding cache"
-            terminus drush $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV -- cr
+          # Set the connection type to GIT and force push
+          terminus connection:set $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV git
+          git push -f pantheon $CIRCLE_BRANCH
+          # Wait upto 240 seconds for the Sync code workflow to finish
+          echo "Waiting for the workflow: Sync code on $PANTHEON_ENV..."
+          terminus build:workflow:wait --max 240 $CIRCLE_PROJECT_REPONAME.$PANTHEON_ENV "Sync code on \"$PANTHEON_ENV\""
```

Comment:
I'm curious, why the max wait of 240 seconds? I would think we would want to always wait until it's finished. If the wait ever times out and we proceed before the event has finished, it could lead to build failure or other issue.

Reply:
You are right that the code would continue after the timeout. The ideal way is to error out when the timeout happens but I can't seems to figure out how to do that with the build plugin. Without setting timeout here, there must be some other timeout settings that will terminate the job eventually but I'm not sure where they are.

Reply:
Well, my recommendation would be to remove the `--max 240` timeout from the command so that it will wait indefinitely for the event to finish. Should we ever encounter an issue where the Pantheon event doesn't finish for some reason and causes the build to be stuck waiting, we can manually stop the CircleCI job. This should be a rare, if ever occurrence, but it seems better to deal with it that way than to risk a build failure if an event takes 300 seconds to run for some reason.